book/integration-service.md

```diff
@@ -0,0 +1,264 @@
+
+  - How to design the tekton pipeline/task to allow the graph to continue to execute [Issue 1684](https://github.com/tektoncd/pipeline/issues/1684) which has spun off a few TEPS to try to mitigate:
+
+    - The [Finally](https://tekton.dev/vault/pipelines-v0.16.3/pipelines/#adding-finally-to-the-pipeline) clause was introduced
+    - [TEP0040](https://github.com/jerop/community/blob/22ad7f6ae9c558ce66d74c1186d3a528cf846677/teps/0040-ignore-step-errors.md) Task Step OnError (Alpha only)
+    - [TEP0050](https://github.com/jerop/community/blob/22ad7f6ae9c558ce66d74c1186d3a528cf846677/teps/0050-ignore-task-failures.md) Ignore Task Failures (Proposed)
+    - [TEP0056](https://github.com/tektoncd/community/blob/main/teps/0056-pipelines-in-pipelines.md) Pipeline in Pipeline (Proposed)
+    - [TEP0049](https://github.com/tektoncd/community/blob/main/teps/0048-task-results-without-results.md) Task Results without Results (Proposed)
+
+  - How to get the results in times of failure. [Issue #3749](https://github.com/tektoncd/pipeline/issues/3749)
+    - [TEP0044](https://github.com/tektoncd/community/blob/main/teps/0044-data-locality-and-pod-overhead-in-pipelines.md) Data Locality and Pod Overhead in Pipelines (proposed)
+
+
+### Proposal
+We should be consistent and continue to design the Test pipelines so that the tasks don’t fail and continue to produce the HACBS_TEST_OUTPUT.
+
+#### Option 1
+leverage the enterprise contract task to run the test validation rego policy against this pipeline.
+
+#### Option 2
+
+If being a part of the Enterprise Contract is NOT possible and not clear. We should still implement the same technology and mechanism we do now with EC. What we can do is extend the IntegrationTestScenario spec to have a testPolicyWhitelist field that a user defines per Scenario. This Policy will actually be a git repo to a data file that can be used by a conftest/opa rego file. 
+
+We would need to
+  - The Test Task Generates the HACBS_TEST_OUPTUT
+    - We may need to expand on the Task Output to accommodate more complex use case than the existing Sanity Tests
+
+  - We define a default test [policy](https://github.com/hacbs-contract/ec-policies/blob/main/policy/release/test.rego) for the Integration Service similar to how EC did. 
+  - We define a data policy format to use with this policy
+
+We submit a TaskRun post pipeline run to validate the Pipeline Run output that includes the HACBS_TEST_OUTPUT
+
```

Comment:
There's that sneaky "we" again - is the integration service handling all of this?

Reply:
Since these Open Questions are already implemented/covered in the above definitions.

Reply:
This section is no longer needed. Done